### PR TITLE
bm16s - Added Via Support,Keymap,Product ID. Fixed Vendor ID

### PR DESCRIPTION
**Pull request by Relocks**
Via Specific Keymap for BM16s along with correction of the Vendor and Product ID.

## Description

- Added necessary files for Via compatability including keymap and rules along with an additional readme for clarification. 

- Changed details within Config to give proper Vendor ID for KPrepublic and a unique Product ID.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/bm16s/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/bm16s/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/bm16s/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  \
+	),
+	[3] = LAYOUT_ortho_4x4(
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, \
+		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  \
```
Comment (fauxpark) [inline keyboards/bm16s/keymaps/via/keymap.c:26]:
```suggestion
	[0] = LAYOUT_ortho_4x4(
		KC_KP_7, KC_KP_8, KC_KP_9, MO(1),
		KC_KP_4, KC_KP_5, KC_KP_6, KC_PMNS,
		KC_KP_1, KC_KP_2, KC_KP_3, KC_PPLS,
		KC_KP_0, KC_PDOT, KC_PCMM, KC_PENT
	),
	[1] = LAYOUT_ortho_4x4(
		RESET,   BL_STEP, KC_TRNS, KC_VOLU,
		BL_TOGG, BL_DEC,  BL_INC,  KC_VOLD,
		RGB_TOG, RGB_MOD, RGB_HUI, KC_MUTE,
		RGB_SAI, RGB_SAD, RGB_HUD, KC_TRNS
	),
	[2] = LAYOUT_ortho_4x4(
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
	),
	[3] = LAYOUT_ortho_4x4(
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
		KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 3]
```diff
@@ -22,5 +22,6 @@
     "[markdown]": {
         "editor.trimAutoWhitespace": false,
         "files.trimTrailingWhitespace": false
-    }
+    },
+    "cmake.configureOnOpen": false
```
Comment (fauxpark) [inline .vscode/settings.json:26]:
```suggestion
    }
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 4]
```diff
@@ -22,5 +22,7 @@
     "[markdown]": {
         "editor.trimAutoWhitespace": false,
         "files.trimTrailingWhitespace": false
+    },
+    "cmake.configureOnOpen": false
```
Comment (fauxpark) [inline .vscode/settings.json:26]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]
Thanks!

**Pull request merged**